api/src/endpoints/postSignin.js

```diff
@@ -0,0 +1,59 @@
+const mongoose = require("mongoose");
+const { usersSchema } = require("../mongooseSchemas");
+const users = mongoose.model("users", usersSchema);
+
+const postSignin = {
+  method: "POST",
+  url: "/api/signin",
+  schema: {
+    body: {
+      type: "object",
+      properties: {
+        u: { type: "string" },
+        a: {
+          type: "object",
+          properties: {
+            o: { type: "string" },
+            te: { type: "string" },
+            t: { type: "string" }
+          }
+        }
+      }
+    }
+  },
+  handler: async (request, reply) => {
+    const signingUser = request.body;
+    if (
+      !signingUser.u ||
+      !signingUser.a ||
+      !signingUser.a.t ||
+      !signingUser.a.te ||
+      !signingUser.a.o
+    ) {
+      console.log(signingUser);
+      reply.code(400).send("Invalid data");
+      return;
+    }
+    const user = new users({
+      u: signingUser.u,
+      a: {
+        t: signingUser.t,
+        te: signingUser.te,
+        o: signingUser.o
+      }
+    });
+
+    try {
+      const savedUser = await user.save();
+      reply.code(200).send(savedUser._id);
+    } catch (e) {
+      if (e.name === "BulkWriteError" && e.code === 11000) {
```

Comment:
didn't work for my db

Reply:
Need to add the script for the unique index.

Reply:
Changed the way we check instead of relaying on unique-index